src/backend/LamaAPI/Application/Common/Interfaces/IApplicationDbContext.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Common.Interfaces
+{
+    public interface IApplicationDbContext
+    {       
+        DbSet<TEntity> Set<TEntity>() where TEntity : class;
+        Task<int> SaveChangesAsync(CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=4794247)